File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let map = new Map();
    let result = 1;
    
    clothes.forEach((e) =>{
        map.set(e[1] , map.has(e[1]) ? map.get(e[1]) + 1  : 1);
    });
    
    for(let [key , value] of map){
        result *= value + 1;
    }
    return result - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    clothes_dict = dict()
    answer = 1
    
    for i in clothes:
        v , k = i
        if k in clothes_dict: clothes_dict[k] += 1
        else: clothes_dict[k] = 1
    
    for i in clothes_dict.values():
        answer *= i + 1

    return answer - 1
